src/main/java/org/mule/service/scheduler/internal/executor/ByCallerThreadGroupPolicy.java

```diff
@@ -61,15 +71,26 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
     ThreadGroup currentThreadGroup = currentThread().getThreadGroup();
 
     if (isWaitGroupThread(targetGroup) && targetGroup == currentThreadGroup) {
+      logRejection(r.toString(), "callerRuns", targetGroup.getName());
       callerRuns.rejectedExecution(r, executor);
     } else if (!isSchedulerThread(currentThreadGroup) || isWaitGroupThread(currentThreadGroup)) {
+      logRejection(r.toString(), "wait", targetGroup.getName());
       // MULE-11460 Make CPU-intensive pool a ForkJoinPool - keep the parallelism when waiting.
       wait.rejectedExecution(r, executor);
     } else {
+      logRejection(r.toString(), "abort", targetGroup.getName());
       abort.rejectedExecution(r, executor);
     }
   }
 
+  private void logRejection(String taskAsString, String strategy, String targetAsString) {
+    if (logRejectionDetails) {
+      LOGGER.warn("Task rejected ({}) from '{}' scheduler: {}", rightPad(strategy, 10), targetAsString, taskAsString);
```

Comment:
Use same logging category for all scheduler reporting enabled via system property, not two different categories.  Personally i wouldn't even use class name, but instead something simpler hierarchical. e.g. `mule.scheduler.usageTrace`.